ipykernel/hookmanager.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""
+A context manager that will register a hook on a
+Display Publisher for storing messages.
+"""
+
+# Copyright (c) IPython Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from io import StringIO
+import sys
+
+from IPython.core.getipython import get_ipython
+
+from .displayhook import ZMQMessageHook
+
+
+class MessageHookFor(object):
```

Comment:
For context managers that are typically treated as functions, I think it's more conventional to use `function_style_name` rather than `ObjectStyleName`, even though it's technically a type.